OneSignalSDK/onesignal/src/main/java/com/onesignal/OSUtils.java

```diff
@@ -369,11 +370,20 @@ int getDeviceType() {
       if (supportsADM())
          return UserState.DEVICE_TYPE_FIREOS;
 
-      if (supportsGooglePush())
+      Context context = OneSignal.appContext;
+      boolean preferHMS = context != null && OSUtils.getManifestMetaBoolean(context, PREFER_HMS_METADATA_NAME);
```

Comment:
nit, this will be a bit more readable if we move these two lines into the `if (supportsFCM && supportsHMS) {` block.